### PR TITLE
chore: modifying edx course filter for course type

### PR DESCRIPTION
**Pull request by alex-sheehan-edx**
jira ticket: https://2u-internal.atlassian.net/browse/ENT-6418
related backend ticket: https://github.com/openedx/enterprise-catalog/pull/511

# description
1) bug fix with url construction for default courses
2) added course type facet to catalog no search results
3) added course type to edx course filter sent to algolia

# For all changes

- [ ] Ensure adequate tests are in place (or reviewed existing tests cover changes)

# Only if submitting a visual change

- [ ] Ensure to attach screenshots
- [ ] Ensure to have UX team confirm screenshots

### CODE REVIEW

File: src/components/catalogs/CatalogSearch.jsx [hunk 1]
```diff
@@ -110,7 +116,11 @@ function CatalogSearch(intl) {
                     filters={courseFilter}
                     facetingAfterDistinct
                   />
-                  <CatalogSearchResults preview contentType={CONTENT_TYPE_COURSE} setNoCourses={setNoCourseResults} />
```
Comment (alex-sheehan-edx) [inline src/components/catalogs/CatalogSearch.jsx:113]: consistency in formatting of components

**Review by kiram15 — APPROVED** [review]
LGTM! 🚀

File: src/data/services/EnterpriseCatalogAPIService.js [hunk 2]
```diff
@@ -28,7 +28,7 @@ class EnterpriseCatalogApiService {
   }
 
   static fetchDefaultCoursesInCatalog(options) {
-    const enterpriseListUrl = `${EnterpriseCatalogApiService.enterpriseCatalogServiceApiUrl}/default_course_set/?${qs.stringify(options)}`;
+    const enterpriseListUrl = `${EnterpriseCatalogApiService.enterpriseCatalogServiceApiUrl}/default_course_set?${qs.stringify(options)}`;
```
Comment (alex-sheehan-edx) [inline src/data/services/EnterpriseCatalogAPIService.js:31]: this is an unrelated bug I found where we were erroneously adding a slash at the end- this was resulting in a 404 on the backend... not sure how this wasn't caught previously

File: src/components/catalogNoResultsDeck/CatalogNoResultsDeck.jsx [hunk 3]
```diff
@@ -37,14 +39,22 @@ function CatalogNoResultsDeck({
...
   useEffect(() => {
     const defaultCoursesRefinements = { enterprise_catalog_query_titles: selectedCatalog, content_type: contentType };
+    if (contentType === CONTENT_TYPE_COURSE) {
+      if (courseType !== null) {
+        defaultCoursesRefinements.course_type = [courseType];
+      } else {
+        // if a course type is not specified, default to edx course content
+        defaultCoursesRefinements.course_type = EDX_COURSES_COURSE_TYPES;
+      }
+    }
```
Comment (marlonkeating) [inline src/components/catalogNoResultsDeck/CatalogNoResultsDeck.jsx]: nit: We can shorten the code here, but it's not necessarily a win in terms of readability, so will leave it to your discretion.
```suggestion
       // if a course type is not specified, default to edx course content 
      defaultCoursesRefinements.course_type = courseType !== null ? [courseType] : EDX_COURSES_COURSE_TYPES;
```

File: src/data/services/EnterpriseCatalogAPIService.js (hunk 2 above)
Comment (marlonkeating) [reply src/data/services/EnterpriseCatalogAPIService.js:31]: Is this something we might include in a unit test?

File: src/components/catalogNoResultsDeck/CatalogNoResultsDeck.jsx (hunk 3 above)
Comment (alex-sheehan-edx) [reply src/components/catalogNoResultsDeck/CatalogNoResultsDeck.jsx]: oh I love the suggestions! Let me think on it.

File: src/data/services/EnterpriseCatalogAPIService.js (hunk 2 above)
Comment (alex-sheehan-edx) [reply src/data/services/EnterpriseCatalogAPIService.js:31]: I think this is worth it's own ticket as there are no tests surrounding the api client. Will write one up!
